### PR TITLE
Update latest refresh test

### PR DESCRIPTION
**Pull request by MonicaisHer**
This PR updates three refresh tests for latest/beta channel. These original jakarta refresh tests will check upgrade from latest/edge to latest/beta. This PR's refresh tests  will check:

test-refresh-config-paths: upgrade from 2.0/stable(2.0.0 - 4) to latest/beta(2.1.0 + 1) or provided snap
test-refresh-services: upgrade from 2.0/stable(2.0.0 - 4) to latest/beta(2.1.0 + 1) or provided snap
test-self-refresh-services: self-upgrade of latest/beta(2.1.0 + 1) or provided snap

```
git clone https://github.com/MonicaisHer/edgex-checkbox-provider.git
cd edgex-checkbox-provider/data/latest
git checkout update-latest-test
time (sudo ./run-all-tests-locally.sh -t test-refresh-config-paths.sh -v)
time (sudo ./run-all-tests-locally.sh -t test-refresh-services.sh -v)
time (sudo ./run-all-tests-locally.sh -t test-self-refresh-services.sh -v)
```
Expect:
```
running single test: test-self-refresh-services.sh ...	PASSED
running single test: test-refresh-services.sh ...	PASSED
running single test: test-self-refresh-services.sh ...	PASSED
```

### CODE REVIEW

**Comment by farshidtz** [comment]
Let's make the refresh be an upgrade:
- from 2.1/stable
- to latest/beta | provided snap

**Review by farshidtz — CHANGES_REQUESTED** [review]
I think some of the new echoed messages after snap install may be problematic. The variable's value can be changed from a different file and those hardcoded message will become invalid. The snap install command should print out the installed version by itself, so we don't have to repeat that.

File: data/utils.sh [hunk 1]
```diff
@@ -148,3 +148,9 @@ snap_wait_port_status()
     fi
 }
 
+print_snap_version()
+{
+    local snap=$1
+    snap list edgexfoundry | sed -n 2p
```
Comment (farshidtz) [inline data/utils.sh]:
```suggestion
    local snap_name=$1
    snap list $snap_name | sed -n 2p
```

File: data/latest/test-refresh-config-paths.sh [hunk 2]
```diff
@@ -12,42 +12,41 @@ SCRIPT_DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" >/dev/null && pwd )"
 source "$SCRIPT_DIR/utils.sh"
 
 DEFAULT_TEST_CHANNEL=${DEFAULT_TEST_CHANNEL:-beta}
+EDGEX_STABLE_CHANNEL="2.0/stable"
```
Comment (farshidtz) [inline data/latest/test-refresh-config-paths.sh]: I think it is better to call this the `EDGEX_PREV_STABLE_CHANNEL` to distinguish `EDGEX_STABLE_CHANNEL` in run-all-tests.

File: data/latest/run-all-tests-locally.sh [hunk 3]
```diff
@@ -112,9 +117,13 @@ snap_remove 2>/dev/null > /dev/null
...
     printf "running single test: %s ..." "$SINGLE_TEST"
+    if [ "$SINGLE_TEST" == "test-refresh-services.sh" ]; then
+        snap_download_stable_and_latest
+    fi
 
-    if [ "$SINGLE_TEST" == "test-refresh-config-paths.sh" ] ||
-       [ "$SINGLE_TEST" == "test-refresh-services.sh" ]; then
+    if [ "$SINGLE_TEST" == "test-refresh-config-paths.sh" ]; then
+        EDGEX_IRELAND_SNAP_FILE=$(snap_download_and_ack edgexfoundry --channel=2.0/stable)
+        export EDGEX_IRELAND_SNAP_FILE
```
Comment (farshidtz) [inline data/latest/run-all-tests-locally.sh]: Call this `EDGEX_PREV_STABLE_SNAP_FILE` to simplify future updates?

File: data/latest/test-refresh-config-paths.sh [hunk 4]
```diff
@@ -58,13 +57,15 @@ notUpgradedFiles=$(grep -R "edgexfoundry/$SNAP_REVISION" | grep -v "raft.db")
      
 popd > /dev/null
 if [ -n "$notUpgradedFiles" ]; then
-    echo "files not upgraded to use \"current\" symlink in config files:"
+    echo "Files not upgraded to use \"current\" symlink in config files:"
     echo "$notUpgradedFiles"
     exit 1
 else
-    echo "no files reference previous snap revision found"
+    echo "No files reference previous snap revision found"
```
Comment (farshidtz) [inline data/latest/test-refresh-config-paths.sh]:
```suggestion
    echo "All config file use the current revision path."
```

**Review by farshidtz — COMMENTED** [review]
Looks good. Just some minor comments. I'll start testing now.

File: data/latest/test-refresh-config-paths.sh (hunk 2 above)
Comment (MonicaisHer) [reply data/latest/test-refresh-config-paths.sh]: I agree. Updated.

File: data/latest/run-all-tests-locally.sh (hunk 3 above)
Comment (MonicaisHer) [reply data/latest/run-all-tests-locally.sh]: Thanks! Updated.

**Review by farshidtz — APPROVED** [review]
Thanks a lot @MonicaisHer.

**Pull request merged**